apps/etterlatte-behandling/src/main/kotlin/grunnlagsendring/doedshendelse/kontrollpunkt/DoedshendelseKontrollpunktService.kt

```diff
@@ -34,23 +38,149 @@ class DoedshendelseKontrollpunktService(
+                }
+            }
+        return kontrollPunkterForRelasjon + fellesKontrollpunkter(hendelse, avdoed, sak)
+    }
+
+    fun fellesKontrollpunkter(
+        hendelse: DoedshendelseInternal,
+        avdoed: PersonDTO,
+        sak: Sak?,
+    ): List<DoedshendelseKontrollpunkt> {
         return listOfNotNull(
-            kontrollerUfoeretrygdBarn(hendelse),
-            kontrollerBarnOgHarBP(hendelse, sak),
             kontrollerAvdoedDoedsdato(avdoed),
-            kontrollerKryssendeYtelse(hendelse),
             kontrollerDNummer(avdoed),
             kontrollerUtvandring(avdoed),
-            kontrollerSamtidigDoedsfall(avdoed, hendelse),
-            kontrollerEksisterendeSak(hendelse, sak),
             kontrollerEksisterendeHendelser(hendelse, sak),
         )
     }
 
-    fun kontrollerUfoeretrygdBarn(doedshendelse: DoedshendelseInternal): DoedshendelseKontrollpunkt.BarnHarUfoereTrygd? {
+    fun kontrollPunkterBarneRelasjon(
+        hendelse: DoedshendelseInternal,
+        avdoed: PersonDTO,
+        sak: Sak?,
+    ): List<DoedshendelseKontrollpunkt> {
+        return listOfNotNull(
+            kontrollerUfoeretrygdBarn(hendelse),
+            kontrollerBarnOgHarBP(hendelse, sak),
+            kontrollerSamtidigDoedsfall(avdoed, hendelse),
+        )
+    }
+
+    fun kontrollPunkterEpsRelasjon(
+        hendelse: DoedshendelseInternal,
+        sak: Sak?,
+        eps: PersonDTO,
+        avdoed: PersonDTO,
+    ): List<DoedshendelseKontrollpunkt> {
+        return listOfNotNull(
+            kontrollerKryssendeYtelseEps(hendelse),
+            kontrollerEksisterendeSakEps(hendelse, sak),
+            kontrollerEpsErDoed(eps),
+            kontrollerEpsHarFylt67Aar(eps),
+            kontrollerSkiltSistefemAarEllerGiftI15(eps, avdoed),
+            kontrollerSkilti5AarMedUkjentGiftemaalStart(eps, avdoed),
+        )
+    }
+
+    private fun kontrollerSkilti5AarMedUkjentGiftemaalStart(
+        eps: PersonDTO,
+        avdoed: PersonDTO,
+    ): DoedshendelseKontrollpunkt.EpsHarVaertSkiltSiste5MedUkjentGiftemaalLengde? {
+        if (eps.sivilstand != null) {
+            val skilt = eps.sivilstand!!.map { it.verdi }.filter { it.sivilstatus == Sivilstatus.SKILT }
+            val skiltMedAvdoed = skilt.filter { it.relatertVedSiviltilstand?.value == avdoed.foedselsnummer.verdi.value }
+            val naa = LocalDate.now()
+            val gift = eps.sivilstand!!.map { it.verdi }.filter { it.sivilstatus == Sivilstatus.GIFT }
+            val giftMedAvdoed = gift.filter { it.relatertVedSiviltilstand?.value == avdoed.foedselsnummer.verdi.value }
+            if (skiltMedAvdoed.isNotEmpty() && giftMedAvdoed.isNotEmpty()) {
+                val skiltsivilstand = skiltMedAvdoed.first()
+                val antallSkilteAar = ChronoUnit.YEARS.between(skiltsivilstand.gyldigFraOgMed, naa).absoluteValue
+                val giftSivilstand = giftMedAvdoed.first()
```

Comment:
Fryktelig tung å lese og et par unødvendige variabler. Hva med noe sånt i stedet? 

```suggestion
            val skiltMedAvdoed = eps.sivilstand!!
                .map(OpplysningDTO<Sivilstand>::verdi)
                .filter { it.sivilstatus == Sivilstatus.SKILT }
                .filter { it.relatertVedSiviltilstand?.value == avdoed.foedselsnummer.verdi.value }

            val giftMedAvdoed = eps.sivilstand!!
                .map(OpplysningDTO<Sivilstand>::verdi)
                .filter { it.sivilstatus == Sivilstatus.GIFT }
                .filter { it.relatertVedSiviltilstand?.value == avdoed.foedselsnummer.verdi.value }

            if (skiltMedAvdoed.isNotEmpty() && giftMedAvdoed.isNotEmpty()) {
                val skiltsivilstand = skiltMedAvdoed.first()
                val antallSkilteAar = ChronoUnit.YEARS.between(skiltsivilstand.gyldigFraOgMed, LocalDate.now()).absoluteValue
                val giftSivilstand = giftMedAvdoed.first()

```